src/main/java/net/fabricmc/loom/providers/MappingsProvider.java

```diff
@@ -27,97 +27,252 @@
 import net.fabricmc.loom.LoomGradleExtension;
 import net.fabricmc.loom.util.Constants;
 import net.fabricmc.loom.util.DependencyProvider;
+import net.fabricmc.loom.util.PathUtils;
 import net.fabricmc.loom.util.Version;
-import net.fabricmc.mappings.Mappings;
+import net.fabricmc.mapping.tree.TinyMappingFactory;
+import net.fabricmc.mapping.tree.TinyTree;
+import net.fabricmc.stitch.Command;
+import net.fabricmc.stitch.commands.CommandMergeTiny;
 import net.fabricmc.stitch.commands.CommandProposeFieldNames;
+import net.fabricmc.stitch.commands.CommandReorderTiny;
+import net.fabricmc.stitch.commands.tinyv2.CommandMergeTinyV2;
+import net.fabricmc.stitch.commands.tinyv2.CommandProposeV2FieldNames;
+import net.fabricmc.stitch.commands.tinyv2.CommandReorderTinyV2;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.tools.ant.util.StringUtils;
 import org.gradle.api.Project;
 
+import java.io.BufferedReader;
 import java.io.File;
-import java.io.FileInputStream;
 import java.io.IOException;
-import java.lang.ref.SoftReference;
+import java.net.URL;
 import java.nio.file.FileSystem;
 import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.function.Consumer;
 
-//TODO fix local mappings
-//TODO possibly use maven for mappings, can fix above at the same time
+import javax.annotation.Nullable;
+
+
 public class MappingsProvider extends DependencyProvider {
-	public MinecraftMappedProvider mappedProvider;
```

Comment:
Please fix the formatting currenly very hard to review

Reply:
Sure thing